vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors_moe.py

```diff
@@ -232,17 +260,43 @@ def apply(
             scoring_func=scoring_func,
             e_score_correction_bias=e_score_correction_bias)
 
-        return fused_experts(x,
-                             layer.w13_weight,
-                             layer.w2_weight,
-                             topk_weights=topk_weights,
-                             topk_ids=topk_ids,
-                             inplace=True,
-                             use_fp8_w8a8=True,
-                             w1_scale=layer.w13_weight_scale,
-                             w2_scale=layer.w2_weight_scale,
-                             a1_scale=layer.w13_input_scale,
-                             a2_scale=layer.w2_input_scale)
+        #TODO should the codepath be decided here?
+        dev_capability = current_platform.get_device_capability().to_int()
+        if dev_capability == 90:
+            from vllm.model_executor.layers.fused_moe import cutlass_moe
+            x_q, x_scale = ops.scaled_fp8_quant(x,
+                                                use_per_token_if_dynamic=False)
+            return cutlass_moe(
```

Comment:
Instead of doing this if/else branching, are we planning to add a layer to select the appropriate kernel? That way we wouldn't have to create parameters that are not required.